### PR TITLE
[yaml] add use-case example

### PR DESCRIPTION
**Pull request by Polber**
Adds 2 simple use-case examples to examples catalog.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Polber** [comment]
R: @liferoad

File: sdks/python/apache_beam/yaml/examples/simple_filter_and_combine.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    - type: WriteToCsv
+      name: WriteOutputFile
+      input: CountNumberSold
+      config:
+        path: output
+
+options:
+  yaml_experimental_features: Combine
+
+# Expected:
```
Comment (liferoad) [inline sdks/python/apache_beam/yaml/examples/simple_filter_and_combine.yaml:54]: [README.md](https://github.com/apache/beam/blob/9add040efc69c516b5462668ac923979cc38a6d0/sdks/python/apache_beam/yaml/examples/README.md) can we update this?

File: sdks/python/apache_beam/yaml/examples/resources/products.csv [hunk 2]
```diff
@@ -0,0 +1,6 @@
+transaction_id,product_name,category,price
+T0012,Headphones,Electronics,59.99
+T5034,Leather Jacket,Apparel,109.99
+T0024,Aluminum Mug,Kitchen,29.99
+T0104,Headphones,Electronics,59.99
+T0302,Monitor,Electronics,249.99
```
Comment (liferoad) [inline sdks/python/apache_beam/yaml/examples/resources/products.csv:6]: do we need to this csv file? can we just create a temp csv for your example?

File: sdks/python/apache_beam/yaml/examples/simple_filter.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# coding=utf-8
```
Comment (liferoad) [inline sdks/python/apache_beam/yaml/examples/simple_filter.yaml:1]: are these yaml files packaged when releasing beam?

Comment (Polber) [reply sdks/python/apache_beam/yaml/examples/simple_filter.yaml:1]: I would assume everything under `apache_beam/` is but I suppose I am not 100% on that

File: sdks/python/apache_beam/yaml/examples/resources/products.csv (hunk 2 above)
Comment (Polber) [reply sdks/python/apache_beam/yaml/examples/resources/products.csv:6]: I think that would defeat the purpose of the example - the point is for it to be runnable as is

File: sdks/python/apache_beam/yaml/examples/simple_filter_and_combine.yaml (hunk 1 above)
Comment (Polber) [reply sdks/python/apache_beam/yaml/examples/simple_filter_and_combine.yaml:54]: Update it how?

Comment (liferoad) [reply sdks/python/apache_beam/yaml/examples/simple_filter_and_combine.yaml:54]: Add this to the list?

File: sdks/python/apache_beam/yaml/examples/simple_filter.yaml (hunk 3 above)
Comment (liferoad) [reply sdks/python/apache_beam/yaml/examples/simple_filter.yaml:1]: we could always create the temp csv file and run the example.

File: sdks/python/apache_beam/yaml/examples/simple_filter_and_combine.yaml (hunk 1 above)
Comment (Polber) [reply sdks/python/apache_beam/yaml/examples/simple_filter_and_combine.yaml:54]: It's not really a list, it is just saying there are examples that highlight specific transforms in the `transforms/` subdirectory and to start with Wordcount in the root directory. All other examples are also in the root directory, but I don't think they need calling out. Their purpose is described in the header of each

File: sdks/python/apache_beam/yaml/examples/resources/products.csv (hunk 2 above)
Comment (Polber) [reply sdks/python/apache_beam/yaml/examples/resources/products.csv]: Fixed by moving csv to `apache-beam-testing`.

File: sdks/python/apache_beam/yaml/examples/simple_filter.yaml (hunk 3 above)
Comment (Polber) [reply sdks/python/apache_beam/yaml/examples/simple_filter.yaml:1]: Also fixed by moving csv to `apache-beam-testing`.

**Comment by Polber** [comment]
R: @damccorm

**Review by liferoad — APPROVED** [review]

**Pull request merged**